lib/ansible/modules/network/basics/uri.py

```diff
@@ -411,8 +411,12 @@ def main():
 
     # Grab all the http headers. Need this hack since passing multi-values is
     # currently a bit ugly. (e.g. headers='{"Content-Type":"application/json"}')
+    deprecations = set()
     for key, value in six.iteritems(module.params):
         if key.startswith("HEADER_"):
+            deprecations.add('Supplying headers via HEADER_* is deprecated and '
```

Comment:
Ok, although I didn't see the logic to de-dupe deprecations, it's done somewhere.  I've removed the `set` and just directly added the deprecation.